store/sqlstore/channel_store.go

```diff
@@ -559,81 +559,42 @@ func (s SqlChannelStore) CreateDirectChannel(user *model.User, otherUser *model.
-		// TODO: This will go away once SaveDirectChannel returns error
-		var invErr *store.ErrInvalidInput
-		var intErr *store.ErrInternal
-		var cErr *store.ErrConflict
-		var ltErr *store.ErrLimitExceeded
-		var appErr *model.AppError
-		if errors.As(err, &invErr) {
-			if invErr.Entity == "Channel" && invErr.Field == "DeleteAt" {
-				return newChannel, model.NewAppError("CreateChannel", "store.sql_channel.save.archived_channel.app_error", nil, "", http.StatusBadRequest)
-			}
-			if invErr.Entity == "Channel" && invErr.Field == "Type" {
-				return newChannel, model.NewAppError("CreateChannel", "store.sql_channel.save.direct_channel.app_error", nil, "", http.StatusBadRequest)
-			}
-			if invErr.Entity == "Channel" && invErr.Field == "Id" {
-				return newChannel, model.NewAppError("SqlChannelStore.Save", "store.sql_channel.save_channel.existing.app_error", nil, "id="+invErr.Value.(string), http.StatusBadRequest)
-			}
-		}
-		if errors.As(err, &intErr) {
-			if intErr.Location == "save_channel" {
-				return newChannel, model.NewAppError("CreateChannel", "store.sql_channel.save_channel.save.app_error", nil, intErr.Error(), http.StatusInternalServerError)
-			}
-			if intErr.Location == "save_channel_count" {
-				return newChannel, model.NewAppError("CreateChannel", "store.sql_channel.save_channel.current_count.app_error", nil, intErr.Error(), http.StatusInternalServerError)
-			}
-		}
-		if errors.As(err, &cErr) {
-			if cErr.Resource == "Channel" {
-				return newChannel, model.NewAppError("CreateChannel", store.CHANNEL_EXISTS_ERROR, nil, cErr.Error(), http.StatusBadRequest)
-			}
-		}
-		if errors.As(err, &ltErr) {
-			if ltErr.What == "channels_per_team" {
-				return newChannel, model.NewAppError("CreateChannel", "store.sql_channel.save_channel.limit.app_error", nil, ltErr.Error(), http.StatusBadRequest)
-			}
-		}
-		if errors.As(err, &appErr) {
-			return nil, appErr
-		}
+		return newChannel, err
 	}
 
 	// Members need new channel ID
 	member1.ChannelId = newChannel.Id
 	member2.ChannelId = newChannel.Id
 
-	var memberSaveErr *model.AppError
 	if member1.UserId != member2.UserId {
-		_, memberSaveErr = s.saveMultipleMembersT(transaction, []*model.ChannelMember{member1, member2})
+		_, err = s.saveMultipleMembersT(transaction, []*model.ChannelMember{member1, member2})
 	} else {
-		_, memberSaveErr = s.saveMemberT(transaction, member2)
+		_, err = s.saveMemberT(transaction, member2)
 	}
-
-	if memberSaveErr != nil {
-		return nil, model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.add_members.app_error", nil, memberSaveErr.Error(), http.StatusInternalServerError)
+	if err != nil {
+		return nil, err
 	}
 
 	if err := transaction.Commit(); err != nil {
-		return nil, model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.commit.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return nil, store.NewErrInternal("commit_transaction", err, "")
```

Comment:
There is any advantage between `NewErrInternal` and use directly errors.New?

Reply:
I wanted to keep the Location of the error as a field so that we don't lose track of what is important. Otherwise we would just have a plain string of location + metadata.

We can change it to errors.New/errors.Wrap if you think keeping the structure is not important. I know that's what was decided on initially. Let me know.